cmd/terminal.go

```diff
@@ -55,31 +59,69 @@ const (
 	// default terminal size
 	defaultTermWidth  = 80
 	defaultTermHeight = 40
+
+	// dummy delay for playback
+	playbackSleep = time.Millisecond * 32
+
+	// cli args
+	argRecord   = "record"
+	argPlayback = "playback"
 )
 
 var terminalCmd = &cobra.Command{
 	Use:   "terminal DEVICE_ID",
 	Short: "Access a device's remote terminal",
-	Args:  cobra.ExactArgs(1),
+	Args:  cobra.RangeArgs(0, 1),
 	Run: func(c *cobra.Command, args []string) {
 		cmd, err := NewTerminalCmd(c, args)
 		CheckErr(err)
 		CheckErr(cmd.Run())
 	},
 }
 
+func init() {
+	terminalCmd.Flags().StringP(argRecord, "", "", "recording file path to save the session to")
+	terminalCmd.Flags().StringP(argPlayback, "", "", "recording file path to playback the session from")
+}
+
 // TerminalCmd handles the terminal command
 type TerminalCmd struct {
-	server      string
-	skipVerify  bool
-	deviceID    string
-	sessionID   string
-	running     bool
-	healthcheck chan int
-	stop        chan struct{}
-	err         error
+	server             string
+	skipVerify         bool
+	deviceID           string
+	sessionID          string
+	running            bool
+	healthcheck        chan int
+	stop               chan struct{}
+	err                error
+	recordFile         string
+	recording          bool
+	stopRecording      chan bool
+	playbackFile       string
+	terminalOutputChan chan []byte
+}
+
+type TerminalRecordingHeader struct {
+	Version  uint8
+	DeviceID [64]byte
+	DateTime [32]byte
```

Comment:
really nitpicking, and not super important, but when I see magic numbers, I always have this urge to do a `const DeviceIDMaxLength`.

Reply:
I'll do that, thanks.